pageserver/src/tenant/timeline.rs

```diff
@@ -3165,9 +3191,8 @@ impl Timeline {
         target_file_size: u64,
         ctx: &RequestContext,
     ) -> Result<CompactLevel0Phase1Result, CompactionError> {
-        let layers = self.layers.read().unwrap();
+        let layers = self.layers.read().await;
         let mut level0_deltas = layers.get_level0_deltas()?;
-        drop(layers);
```

Comment:
I'll need to understand why this needed to be dropped.

Reply:
This thread in re: 24df184a.

Oki the problem becomes `all_keys_iter` and `all_values_iter` is not Send. And that is not Send because they hold the internal to delta layer RwLockGuard.

I don't think this is so controversial.